src/dmd/doc.d

```diff
@@ -1964,6 +1992,102 @@ Lno:
     return i;
 }
 
+/****************************************************
+ * Process Markdown emphasis
+ * Params:
+ *  buf =               an OutputBuffer containing the DDoc
+ *  i =                 the index within `buf` to end the Markdown heading at. Its value changes if any emphasis was replaced.
+ *  loc =               the current location within the file
+ *  inlineDelimiters =  the collection of delimiters found within a paragraph. When this function returns its length will be reduced to `downToLevel`.
+ *  downToLevel =       the length within `inlineDelimiters`` to reduce emphasis to
+ * Returns: the amount `i` was moved
+ */
+private size_t replaceMarkdownEmphasis(OutBuffer *buf, ref size_t i, Loc loc, ref MarkdownDelimiter[] inlineDelimiters, int downToLevel = 0)
+{
+    size_t replaceEmphasisPair(ref MarkdownDelimiter start, ref MarkdownDelimiter end)
+    {
+        int count = end.count == 1 ? 1 : 2;
+        if (start.count < count)
+            count = start.count;
+
+        size_t iStart = start.iStart;
+        size_t iEnd = end.iStart;
+        end.count -= count;
+        start.count -= count;
+        iStart += start.count;
+
+        if (!start.count)
+            start.type = 0;
+        if (!end.count)
+            end.type = 0;
+
+        if (global.params.vmarkdown)
+        {
+            buf.data[iEnd] = '\0';
+            char *s = cast(char*)(buf.data + iStart + count);
```

Comment:
You can't delete line 2026 without messing up the output, because `message` takes a C-style zero-terminated string. Luckily `buf.data[iEnd]` gets overwritten by the emphasis replacements so setting it to `\0` here doesn't hurt anything.